File: Development/WoodLotBackEnd/src/main/java/it/unisa/WoodLot/model/entity/Albero.java
package it.unisa.WoodLot.model.entity;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

/**
 * Classe che modella un albero
 *
 * @author Alessia Ture
 */
@Data
@Entity
@Table(name = "albero")
public class Albero implements Serializable {

    @Id
    @Column(name = "nome")
    private String nome;

    @Column(name = "specie_scientifica", nullable = false)
    private String specieScientifica;

    @Column(name = "descrizione_breve", nullable = false, columnDefinition = "TEXT")
    private String descrizioneBreve;

    @Column(name = "descrizione", nullable = false, columnDefinition = "TEXT")
    private String descrizione;

    @Column(name = "prezzo", nullable = false)
    private double prezzo;

    @Column(name = "CO2", nullable = false)
    private int anidrideCarbonicaAssorbita;

    @Column(name = "salvaguardia", nullable = false)
    private int salvaguardia;

    @OneToOne()
    @JoinColumn(name = "categoria", referencedColumnName = "nome")
    private Categoria categoria;

    @OneToOne()
    @JoinColumn(name = "paese_origine", referencedColumnName = "nome")
    private PaeseOrigine paeseOrigine;

    @ManyToMany
    @JoinColumn(name = "uso_locale", referencedColumnName = "nome")
    private List<UsoLocale> usiLocali;

    @ManyToMany
    @JoinColumn(name = "beneficio", referencedColumnName = "nome")
    private List<Beneficio> benefici;
}

File: Development/WoodLotFrontEnd/src/app/entita/albero/albero.ts
import {UsoLocale} from "../usoLocale/uso-locale";
import {Paese} from "../paese/paese";
import {Beneficio} from "../beneficio/beneficio";
import {Categoria} from "../categoria/categoria";

export class Albero {

  public prezzo: number
  public anidrideCarbonicaAssorbita: number
  public nome: String
  public specieScientifica: String | undefined
  public descrizioneBreve: String | undefined
  public descrizione: String | undefined
  public salvaguardia: number | undefined
  public usiLocali: Array<UsoLocale> | undefined
  public benefici: Array<Beneficio> | undefined
  public categoria: Categoria | undefined
  public paeseOrigine: Paese | undefined

  constructor(prezzo: number, co2: number, nome: String,
              nomeScientifico?: String, breveDescrizione?: String, descrizione?: String, salvaguardiaAmbientale?: number,
              usiLocali?: Array<UsoLocale>, benefici?: Array<Beneficio>, categoria?: Categoria, paeseOrigine?: Paese) {
    this.prezzo = prezzo;
    this.anidrideCarbonicaAssorbita = co2;
    this.nome = nome;
    this.specieScientifica = nomeScientifico;
    this.descrizioneBreve = breveDescrizione;
    this.descrizione = descrizione;
    this.salvaguardia = salvaguardiaAmbientale;
    this.usiLocali = usiLocali;
    this.benefici = benefici;
    this.categoria = categoria;
    this.paeseOrigine = paeseOrigine;
  }

}
